agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -158,7 +187,22 @@ func (cs *CredentialSpecResource) handleCredentialspecFile(credentialspec string
 		return errors.New("invalid credentialspec file specification")
 	}
 
-	dockerHostconfigSecOptCredSpec := strings.Replace(credentialspec, "credentialspec:", "credentialspec=", 1)
+	if strings.HasPrefix(credentialspec, "credentialspecdomainless:") {
+		// We need to copy the customers local file in order to avoid modifying their credentialspec
+		credSpecFile, err = cs.copyCredentialSpecLocalFile(credentialspec)
```

Comment:
We should do this in three steps.

1. Read from cx file into memory map[string]int
2. inject the missing domainless params
3. write to file in docker root

Reply:
done